tests/scripts/thread-cert/pktverify/addrs.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2019, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+import ipaddress
+from typing import Union
+
+from pktverify.bytes import Bytes
+
+
+class EthAddr(Bytes):
+
+    def __init__(self, addr: Union[str, bytearray, 'Bytes']):
```

Comment:
Does `'Bytes'` here represent a type? What's the function of it here? It's just my question. Why isn't `Bytes` used instead.

Reply:
That a fair question. 

The benefit of using `Bytes` to represent a byte array instead of just using builtin classes `bytearray` or `bytes` is that it allows us to write conditions in a easier way. 
Bytes automatically convert strings when comparing, which allows us to write filters like:
```python
p.wpan.src64 == "00:11:22:33:44:55:66:77"
```
Note `p.wpan.src64` returns an `ExtAddr` value which is a subclass of `Bytes`. 

It also allows us to compare IP6 addresses easily. 
```python
p.ip6.src[:8] == "fd00:0d8b:0000:0000"
```

Types like `ExtAddr`, `EthAddr`, `Ipv6Addr` are subclasses of `Bytes`. They are used for layer fields of known types.

Reply:
Thanks very much about the explanation. I actually mean in
```
def __init__(self, addr: Union[str, bytearray, 'Bytes']):
```
Why should it need `'` around `Bytes`? Could it be like:
```
def __init__(self, addr: Union[str, bytearray, Bytes]):
```

Reply:
Oh. Good question. 

In short, we can't use `Bytes` until Python 3.7+, so we have to use `'Bytes'` because Python 3.6 is supported. 
Please check https://stackoverflow.com/a/33533514 for more details on this.